File: kotlin/src/main/kotlin/991.BrokenCalculator.kt
import org.junit.Assert.assertEquals
import org.junit.Test
//https://leetcode.com/problems/broken-calculator/


class BrokenCalculator{
  fun brokenCalc(X: Int, Y: Int): Int {
    var cnt = 0
    var y = Y
    while(y > X){
      if(y and 1 == 1){
        y += 1
        cnt ++
      }
      cnt ++
      y = y ushr 1
    }
    return cnt + X-y
  }

  @Test
  fun test1(){
    assertEquals(2, brokenCalc(2,3))
  }

  @Test
  fun test2(){
    assertEquals(2, brokenCalc(5,8))
  }

  @Test
  fun test3(){
    assertEquals(3, brokenCalc(3,10))
  }

  @Test
  fun test4(){
    assertEquals(1023, brokenCalc(1024,1))
  }
}

File: python/991_BrokenCalculator.py
from unittest import TestCase
# https://leetcode.com/problems/broken-calculator/


class BrokenCalculator(TestCase):
    def brokenCalc(self, X: 'int', Y: 'int') -> 'int':
        cnt = 0
        while Y > X:
            cnt += 1
            if Y & 1:
                Y += 1
            else:
                Y >>= 1
        return cnt + X-Y

    def test1(self):
        self.assertEqual(2, self.brokenCalc(2, 3))

    def test2(self):
        self.assertEqual(2, self.brokenCalc(5, 8))

    def test3(self):
        self.assertEqual(3, self.brokenCalc(3, 10))

    def test4(self):
        self.assertEqual(1023, self.brokenCalc(1024, 1))


